File: app/src/main/java/workshop1024/com/xproject/view/dialog/SubmitMessageDialog.java
package workshop1024.com.xproject.view.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.view.View;

import workshop1024.com.xproject.R;
import workshop1024.com.xproject.databinding.SubmitmessageDialogBinding;

public class SubmitMessageDialog extends DialogFragment {
    public static final String TAG = "SubmitMessageDialog";
    private SubmitMessageDialog mSubmitMessageDialog;

    private SubmitMessageDialogListener mSubmitMessageDialogListener;
    private SubmitmessageDialogBinding mSubmitmessageDialogBinding;

    public static SubmitMessageDialog newInstance() {
        SubmitMessageDialog submitMessageDialog = new SubmitMessageDialog();
        return submitMessageDialog;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mSubmitMessageDialog = this;
        try {
            mSubmitMessageDialogListener = (SubmitMessageDialogListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() + " must implement SubmitMessageDialogListener");
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        mSubmitmessageDialogBinding = DataBindingUtil.inflate(getActivity().getLayoutInflater(),
                R.layout.submitmessage_dialog, null, false);
        mSubmitmessageDialogBinding.setSubmitMessageHandlers(new SubmitMessageHandlers());
        return new AlertDialog.Builder(getActivity()).setView(mSubmitmessageDialogBinding.getRoot()).create();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mSubmitMessageDialogListener.cancelButtonClick(this);
    }

    public interface SubmitMessageDialogListener {
        void cancelButtonClick(DialogFragment dialogFragment);

        void submitButtonClick(DialogFragment dialogFragment, String messageConent);
    }

    public class SubmitMessageHandlers {
        public void onClickCancel(View view) {
            mSubmitMessageDialogListener.cancelButtonClick(mSubmitMessageDialog);
        }

        public void onClickSubmit(View view) {
            mSubmitMessageDialogListener.submitButtonClick(mSubmitMessageDialog, mSubmitmessageDialogBinding.
                    submitmessageEdittextMessage.getText().toString());
        }
    }
}

File: android/app/src/main/java/workshop1024/com/xproject/view/dialog/SubmitMessageDialog.kt
package workshop1024.com.xproject.view.dialog

import android.app.Dialog
import android.content.Context
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.view.View
import workshop1024.com.xproject.R
import workshop1024.com.xproject.databinding.SubmitmessageDialogBinding

class SubmitMessageDialog : DialogFragment() {
    private lateinit var mSubmitMessageDialog: SubmitMessageDialog

    private var mSubmitMessageDialogListener: SubmitMessageDialogListener? = null
    private lateinit var mSubmitmessageDialogBinding: SubmitmessageDialogBinding

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        mSubmitMessageDialog = this
        try {
            mSubmitMessageDialogListener = context as SubmitMessageDialogListener?
        } catch (e: ClassCastException) {
            throw ClassCastException(context!!.toString() + " must implement SubmitMessageDialogListener")
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        mSubmitmessageDialogBinding = DataBindingUtil.inflate(activity!!.layoutInflater,
                R.layout.submitmessage_dialog, null, false)
        mSubmitmessageDialogBinding.submitMessageHandlers = SubmitMessageHandlers()
        return AlertDialog.Builder(activity!!).setView(mSubmitmessageDialogBinding.root).create()
    }

    override fun onDetach() {
        super.onDetach()
        mSubmitMessageDialogListener!!.cancelButtonClick(this)
    }


    interface SubmitMessageDialogListener {
        fun cancelButtonClick(dialogFragment: DialogFragment)

        fun submitButtonClick(dialogFragment: DialogFragment, messageConent: String)
    }

    inner class SubmitMessageHandlers {
        fun onClickCancel(view: View) {
            mSubmitMessageDialogListener!!.cancelButtonClick(mSubmitMessageDialog)
        }

        fun onClickSubmit(view: View) {
            mSubmitMessageDialogListener!!.submitButtonClick(mSubmitMessageDialog, mSubmitmessageDialogBinding.submitmessageEdittextMessage.text.toString())
        }
    }

    companion object {
        const val TAG = "SubmitMessageDialog"

        fun newInstance(): SubmitMessageDialog {
            return SubmitMessageDialog()
        }

    }
}